sensor/common/compliance/node_inventory_handler_test.go

```diff
@@ -136,6 +139,59 @@ func (s *NodeInventoryHandlerTestSuite) TestHandlerStopIgnoresError() {
 	s.NoError(h.Stopped().Wait())
 }
 
+// This test simulates a running Sensor loosing connection to Central, followed by a reconnect.
+// As soon as Sensor enters offline mode, it should send NACKs to Compliance.
+// In online mode, inventories are forwarded to Central, which responds with an ACK, that is passed to Compliance.
+func (s *NodeInventoryHandlerTestSuite) TestHandlerOfflineACKNACK() {
```

Comment:
What do you think about having a state struct like: 

```golang
type testState struct {
    event common.SensorComponentEvent
    expectedACKCount int
    expectedNACKCount int
}
```

Then you can have the body of the test something like this:

```golang
	ch, producer := make(chan *storage.NodeInventory) // we don't need the producer just the channel
	defer close(ch)
	h := NewNodeInventoryHandler(ch, &mockAlwaysHitNodeIDMatcher{})
	s.NoError(h.Start())
	states := []testState{
		{
			event: common.SensorComponentEventCentralReachable,
			ackCount: 1,
			nackCount: 0,
		},
		// then one for offline and another one for online
	}
	for i, state := range states {
		h.Notify(state.event)
		ch <- fakeNodeInventory(fmt.Sprintf("Node-%d", i))
		if state.event == common.SensorComponentEventCentralReachable {
			mockCentralAck(h, s)
		}
		stats := consumeAndCountCompliance(h.ComplianceC(), 1)
		s.NoError(stats.sc.Stopped().Wait())
		s.Equal(state.expectedACKCount, stats.ACKCount)
		s.Equal(state.expectedNACKCount, stats.NACKCount)
	}
    // stop the inventory handler.
```

Reply:
That would work, and while it looks cleaner, I'm not sure if I like it more.
At least for me, this structure makes it harder to understand what is happening (start with online mode, then switch to offline, and back to online).
That said, I'm aware that my approach is more clumsy and suffers from code repition, so there's advantages to both of them

Reply:
Fair enough. I've changed the test, please TAL :)